src/BrainPass/BrainPass.sol

```diff
@@ -0,0 +1,343 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "openzeppelin-contracts/contracts/token/ERC721/ERC721.sol";
+import {Counters} from "openzeppelin-contracts/contracts/utils/Counters.sol";
+import {Owned} from "solmate/auth/Owned.sol";
+import {SafeMath} from "openzeppelin-contracts/contracts/utils/math/SafeMath.sol";
+
+interface IERC20 {
+    function transferFrom(
+        address from,
+        address to,
+        uint256 amount
+    ) external returns (bool);
+}
+
+/// @title BRAIN Pass NFT
+/// @author Oleanji
+/// @notice A pass for IQ Wiki Editors
+
+contract BrainPassCollectibles is ERC721, Owned {
+    /// -----------------------------------------------------------------------
+    /// Errors
+    /// -----------------------------------------------------------------------
+
+    error MintingPaymentFailed();
+    error IncreseTimePaymentFailed();
+    error AlreadyMintedThisPass();
+    error NotTheOwnerOfThisNft();
+    error InvalidMaxTokensForAPass();
+    error PassTypeNotFound();
+    error PassMaxSupplyReached();
+    error NoEtherLeftToWithdraw();
+    error TransferFailed();
+    error DurationNotInTimeFrame();
+
+    /// -----------------------------------------------------------------------
+    ///  Inheritances
+    /// -----------------------------------------------------------------------
+
+    using SafeMath for uint256;
+    using Counters for Counters.Counter;
+
+    /// -----------------------------------------------------------------------
+    /// Structs
+    /// -----------------------------------------------------------------------
+
+    struct UserPassItem {
+        uint256 tokenId;
+        uint256 passId;
+        uint256 startTimestamp;
+        uint256 endTimestamp;
+    }
+
+    struct PassType {
+        uint256 passId;
+        string name;
+        uint256 pricePerDay;
+        string tokenURI;
+        uint256 maxTokens;
+        uint256 discount;
+        uint256 lastTokenIdMinted;
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Mappings
+    /// -----------------------------------------------------------------------
+
+    mapping(uint256 => PassType) public passTypes;
+    mapping(address => mapping(uint256 => UserPassItem))
+        public addressToNFTPass;
+    mapping(uint256 => mapping(address => UserPassItem)) internal ownerOfToken;
+
+    /// -----------------------------------------------------------------------
+    /// Constant
+    /// -----------------------------------------------------------------------
+
+    address public immutable iqToken;
+    uint256 immutable SECONDS_IN_A_DAY = 86400;
```

Comment:
fyi: solidity has time units. you can use 1 days ( https://docs.soliditylang.org/en/v0.8.14/units-and-global-variables.html#time-units )